mailboxes/mailboxes-lib.pl

```diff
@@ -1306,59 +1324,238 @@ sub parse_calendar_file
-		# Start a new event
+# Lines processor
+my $process_line = sub  {
+	my ($line) = @_;
+	# Start a new event
+	if ($line =~ /^BEGIN:VEVENT/) {
 		%event = ();
+		$event{'description'} = [ ];
+		$event{'attendees'} = [ ];
 		}
-	elsif (/^END:VEVENT/) {
-		# Convert times using the timezone
-		if ($event{'dtstart'} && $event{'tzid'}) {
-			my $adjusted_start = $adjust_time_with_timezone->($event{'dtstart'}, $event{'tzid'});
-			$event{'dtstart_local_timestamp'} = $adjusted_start->{'timestamp'};
-			$event{'dtstart_local_date'} = &make_date($event{'dtstart_local_timestamp'});
+	# Convert times using the timezone
+	elsif ($line =~ /^END:VEVENT/) {
+		# Local timezone
+		$event{'tzid_local'} =
+			DateTime::TimeZone->new(name => 'local')->name();
+		$event{'tzid'} = 'UTC', $event{'tzid_missing'} = 1
+			if (!$event{'tzid'});
+		# Adjust times with timezone
+		my ($adjusted_start, $adjusted_end);
+		$event{'tzid'} = $timezone_map{$event{'tzid'}} || $event{'tzid'};
+		# Add single start/end time
+		if ($event{'dtstart'}) {
+			$adjusted_start =
+				$adjust_time_with_timezone->($event{'dtstart'},
+							     $event{'tzid'});
+			$event{'dtstart_timestamp'} = $adjusted_start->{'timestamp'};
+			my $dtstart_date =
+				&make_date($event{'dtstart_timestamp'},
+					{ tz => $event{'tzid'} });
+			$event{'dtstart_date'} =
+				"$dtstart_date->{'short'} ".
+					"$dtstart_date->{'timeshort'}"; 
+			$event{'dtstart_local_timestamp'} =
+				$adjusted_start->{'timestamp_local'};
+			$event{'dtstart_local_date'} =
+				&make_date($event{'dtstart_local_timestamp'});
 			}
-		if ($event{'dtend'} && $event{'tzid'}) {
-			my $adjusted_end = $adjust_time_with_timezone->($event{'dtend'}, $event{'tzid'});
-			$event{'dtend_local_timestamp'} = $adjusted_end->{'timestamp'};
-			$event{'dtend_local_date'} = &make_date($event{'dtend_local_timestamp'});
+		if ($event{'dtend'}) {
+			$adjusted_end =
+				$adjust_time_with_timezone->($event{'dtend'},
+							     $event{'tzid'});
+			$event{'dtend_timestamp'} = $adjusted_end->{'timestamp'};
+			my $dtend_date = &make_date($event{'dtend_timestamp'},
+				{ tz => $event{'tzid'} });
+			$event{'dtend_date'} =
+				"$dtend_date->{'short'} ".
+					"$dtend_date->{'timeshort'}";
+			$event{'dtend_local_timestamp'} =
+				$adjusted_end->{'timestamp_local'};
+			$event{'dtend_local_date'} =
+				&make_date($event{'dtend_local_timestamp'});
+			}
+		if ($event{'dtstart'} && $event{'dtend'}) {
+			# Try to add local 'when (period)'
+			my $dtstart_local_obj =
+			    $event{'_obj_dtstart_local_time'} =
+			        make_date($event{'dtstart_local_timestamp'},
+					{ _ });
+			my $dtend_local_obj =
+			    $event{'_obj_dtend_local_time'} =
+			        make_date($event{'dtend_local_timestamp'},
+					{ _ });
+			# Build when local, e.g.:
+			# Tue Jun 04, 2024 04:30 PM – 05:15
+			# PM (Asia/Nicosia +0300)
+			# or
+			# Tue Jun 04, 2024 04:30 PM – Wed Jun 05, 2024 01:15
+			# AM (Asia/Nicosia +0300)
+			$event{'dtwhen_local'} =
```

Comment:
There's a lot of indentation here! Can you split this out into a separate function?

Reply:
> There's a lot of indentation here! Can you split this out into a separate function?

It would rase complexity, and I'd prefer not to do it, as it fits 80 char width.

Reply:
Okay, actually there was a way to reduce indentation. Fixed now.